onchain/exporter-lib/Exporter/Hash.hs

```diff
@@ -0,0 +1,75 @@
+{-# LANGUAGE RecordWildCards #-}
+module Exporter.Hash (HasScriptHash(scriptHashOf),showHash) where
+
+import Plutarch.Api.V1 ( scriptHash )
+import Plutarch.Script ( Script(Script) )
+import Ply (
+  ScriptVersion(..),
+  TypedScriptEnvelope (
+    TypedScriptEnvelope,
+    tsDescription,
+    tsParamTypes,
+    tsRole,
+    tsScript,
+    tsVersion
+  ),
+ )
+import qualified Data.Text as T
+import Data.ByteArray (convert)
+import qualified Data.ByteString.Base16 as Base16
+import Data.ByteString.Short (fromShort)
+import Data.ByteString (ByteString)
+import qualified Data.Text.Encoding as T
+import PlutusLedgerApi.V1 (ScriptHash(..))
+import PlutusTx.Prelude (toBuiltin, fromBuiltin, BuiltinByteString)
+import PlutusLedgerApi.Common ( serialiseCompiledCode )
+import Crypto.Hash (hashWith)
+import Crypto.Hash.Algorithms (
+  Blake2b_224 (Blake2b_224),
+  HashAlgorithm,
+ )
+import Exporter.Types
+    ( PlutusEnvelope(..), Some(MkSome), ScriptEnvelope(..) )
+
+class HasScriptHash a where
+  scriptHashOf :: a -> ScriptHash
+
+instance HasScriptHash TypedScriptEnvelope where
+  scriptHashOf TypedScriptEnvelope{..} = scriptHash (Script tsScript)
+
+instance HasScriptHash PlutusEnvelope where
```

Comment:
We really shouldn't hardcode both the hashing algorithm and the byte prefixes. Isn't there some instance calculating e2e hash? Only in cardano-api?